### PR TITLE
feat: add redstone node

### PR DESCRIPTION
**Pull request by cehali**

### CODE REVIEW

**Comment by cehali** [comment]
Github actions for monorepo based on https://github.com/int128/typescript-actions-monorepo
Typescript tsconfig.json files based on https://dev.to/mxro/the-ultimate-guide-to-typescript-monorepos-5ap7

File: packages/protocol/src/data-package/SignedDataPackage.ts [hunk 1]
```diff
@@ -14,6 +14,14 @@ import { DataPackage, DataPackagePlainObj } from "./DataPackage";
 
 export interface SignedDataPackagePlainObj extends DataPackagePlainObj {
   signature: string; // base64-encoded joined signature
+  signerAddress: string;
```
Comment (hatskier) [inline packages/protocol/src/data-package/SignedDataPackage.ts:17]: Can we please create this extended interface with non-core data (`signerAddress`, `permawebTx`, `provider`, etc.) in the redstone-node code as it shouldn't be a part of the protocol?

File: packages/protocol/src/data-package/SignedDataPackage.ts [hunk 2]
```diff
@@ -59,9 +67,12 @@ export class SignedDataPackage extends Serializable {
 
   toObj(): SignedDataPackagePlainObj {
     const signatureHex = this.serializeSignatureToHex();
+    const signerAddressAsBytes = this.recoverSignerPublicKey();
```
Comment (hatskier) [inline packages/protocol/src/data-package/SignedDataPackage.ts:70]: Let's not recoverSignerPublicKey in this function, it is not neccessary always for serialization to plain object, but significantly consumes CPU resources

File: packages/protocol/src/data-point/StringDataPoint.ts [hunk 3]
```diff
@@ -7,4 +7,11 @@ export class StringDataPoint extends DataPoint {
     const valueBytes = toUtf8Bytes(value);
     super(symbol, valueBytes);
   }
+
+  serializeToBroadcast() {
```
Comment (hatskier) [inline packages/protocol/src/data-point/StringDataPoint.ts:11]: Shouldn't we remove it?

**Review by hatskier — CHANGES_REQUESTED** [review]

File: packages/protocol/src/data-package/SignedDataPackage.ts (hunk 2 above)
Comment (cehali) [reply packages/protocol/src/data-package/SignedDataPackage.ts]: Let's talk about it tomorrow

**Review by hatskier — APPROVED** [review]

**Pull request merged**